### PR TITLE
Improved wallet UI: main screen

### PR DESCRIPTION
**Pull request by jeluard**
### Summary:

Finished wallet main screen UI.

In particular, following issues are fixed:

* https://github.com/status-im/status-react/issues/1666
* https://github.com/status-im/status-react/issues/1721

Asset lists also implements pull-to-refresh paradigm.

status: ready

### CODE REVIEW

File: .env.jenkins [hunk 1]
```diff
@@ -1,2 +1,2 @@
 TESTFAIRY_ENABLED=1
-WALLET_WIP_ENABLED=1
\ No newline at end of file
+WALLET_WIP_ENABLED=0
```
Comment (rasom) [inline .env.jenkins]: Why is wallet disabled in this PR?

Comment (jeluard) [reply .env.jenkins]: Wallet is now always enabled. WIP features are not enabled.

Comment (rasom) [reply .env.jenkins]: But i suppose we want to have WIP features in Jenkins' builds anyway. Or i misunderstand something?

Comment (oskarth) [reply .env.jenkins]: I think WIP features should be in Jenkins too

File: src/status_im/components/icons/vector_icons.cljs [hunk 2]
```diff
@@ -24,66 +24,71 @@
...
-            :white common/color-white
-            :red common/icon-red-color
-            common/icon-dark-color))))]))
+(def default-viewbox {:width 24 :height 24 :viewBox "0 0 24 24"})
+
+(defn icon
+  ([name] (icon name nil))
+  ([name {:keys [color container-style style]}]
+   [rn/view container-style
+    (if-let [icon (get icons (normalize-property-name name))]
```
Comment (oskarth) [inline src/status_im/components/icons/vector_icons.cljs]: Is the shadowing of icon on purpose? Can we disambiguate them? Also not 100% clear on how this works, with `icon` fn calling `icon` again. Could you clarify code or perhaps write a short comment?

File: src/status_im/components/list/views.cljs [hunk 3]
```diff
@@ -97,15 +104,19 @@
 (defn- default-render-section-header [m]
   [rn/text {:style lst/section-header} (:title m)])
 
+(defn- wrap-per-section-render-fn [v]
+  (map #(if-let [f (:render-fn %)] (assoc (dissoc % :render-fn) :renderItem f) %) v))
```
Comment (oskarth) [inline src/status_im/components/list/views.cljs]: For anon function with multiple usage of %, can we give this variable a descriptive name instead? E.g. what is `v` and what is an item in `v`? A bit hard to parse.

File: src/status_im/components/toolbar_new/view.cljs [hunk 4]
```diff
@@ -1,33 +1,114 @@
 (ns status-im.components.toolbar-new.view
-  (:require [re-frame.core :refer [subscribe dispatch]]
+  (:require [reagent.core :as r]
+            [re-frame.core :as rf]
             [status-im.components.react :as rn]
-            [status-im.components.sync-state.gradient :refer [sync-state-gradient-view]]
+            [status-im.components.sync-state.gradient :as sync-state-gradient-view]
```
Comment (oskarth) [inline src/status_im/components/toolbar_new/view.cljs:5]: A bit odd that it is `sync-state-gradient-view/sync-state-gradient-view`- can we clarify what the namespace and function is responsible for? E.g. something like `sync-state/gradient-view` would make more sense to me (without looking at the code in detail).

File: src/status_im/components/toolbar_new/view.cljs [hunk 5]
```diff
@@ -1,33 +1,114 @@
...
             [status-im.components.styles :as st]
-            [status-im.components.context-menu :refer [context-menu]]
+            [status-im.components.context-menu :as context-menu]
```
Comment (oskarth) [inline src/status_im/components/toolbar_new/view.cljs:7]: Same here as above. Perhaps this is out of scope of PR.

File: src/status_im/components/toolbar_new/view.cljs [hunk 6]
```diff
@@ -1,33 +1,114 @@
...
+             blank-action
+
+             options
+             [option-actions icon icon-opts options]
+
+             :else
+             [icon-action icon icon-opts handler])
+       {:key (str "action-" image)}))])
+
+(defn toolbar2
```
Comment (oskarth) [inline src/status_im/components/toolbar_new/view.cljs:88]: What is toolbar vs toolbar2? Can we give a more descriptive name or a comment that explains when we use on over the other, and what the main difference is?

File: src/status_im/components/icons/vector_icons.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/components/icons/vector_icons.cljs]: Right the shadowing is unfortunate. Nested `icon` is actually a function.

File: src/status_im/components/list/views.cljs (hunk 3 above)
Comment (jeluard) [reply src/status_im/components/list/views.cljs]: Sure

**Comment by flexsurfer** [comment]
wow, huge one:)

File: src/status_im/components/icons/vector_icons.cljs [hunk 7]
```diff
@@ -24,66 +24,71 @@
 (def defs (get-class "Defs"))
 
 (def icons {:icons/chats               (slurp-svg "./resources/icons/bottom/chats_gray.svg")
-            :icons/chats_active        (slurp-svg "./resources/icons/bottom/chats_active.svg")
+            :icons/chats-active        (slurp-svg "./resources/icons/bottom/chats_active.svg")
```
Comment (flexsurfer) [inline src/status_im/components/icons/vector_icons.cljs:27]: thanks :)

File: src/status_im/components/toolbar_new/view.cljs (hunk 4 above)
Comment (jeluard) [reply src/status_im/components/toolbar_new/view.cljs:5]: True. This is a general problem with `components`.

File: src/status_im/components/list/views.cljs [hunk 8]
```diff
@@ -38,12 +38,19 @@
...
-
-(defn item-image [source]
-  [rn/image {:source source
-             :style  lst/item-image}])
+(defn item-icon
+  ([k] (item-icon k nil))
+  ([k {:keys [style icon-opts]}]
+   [rn/view {:style lst/item-image-wrapper}
+    [rn/view {:style style}
+     [vi/icon k (merge icon-opts {:style lst/item-image})]]])) ;; TODO
```
Comment (flexsurfer) [inline src/status_im/components/list/views.cljs]: TODO?

File: src/status_im/components/toolbar_new/view.cljs (hunk 6 above)
Comment (jeluard) [reply src/status_im/components/toolbar_new/view.cljs:88]: `toolbar2` is the more general `toolbar`. I have not yet migrated everything to `toolbar2` but it will replace `toolbar`

Comment (flexsurfer) [reply src/status_im/components/toolbar_new/view.cljs:88]: we have toolbar.toolbar, toolbar_new.toolbar and toolbar_new.toolbar2 :)

File: src/status_im/ui/screens/accounts/views.cljs [hunk 9]
```diff
@@ -47,15 +47,15 @@
...
                      st/accounts-action-button)]
     [common/separator st/accounts-separator st/accounts-separator-wrapper]
     [action-button (merge
-                     {:label (i18n/label :t/recover-access)
-                      :icon [:icons/dots_horizontal
-                             {:color :white}]
-                      :on-press #(dispatch [:navigate-to :recover])}
+                     {:label     (i18n/label :t/recover-access)
+                      :icon      :icons/dots-horizontal
+                      :icon-opts {:color :white}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/accounts/views.cljs:59]: yeah, that's better

File: src/status_im/ui/screens/discover/views/search_results.cljs [hunk 10]
```diff
@@ -49,7 +49,7 @@
        (if (empty? discoveries)
          [view st/empty-view
           ;; todo change icon
```
Comment (flexsurfer) [inline src/status_im/ui/screens/discover/views/search_results.cljs]: we can remove this todo we have designers

File: src/status_im/ui/screens/wallet/main/views.cljs [hunk 11]
```diff
@@ -7,102 +7,138 @@
...
      :icons/dropdown
-     {:container-style st/toolbar-title-icon
+     {:container-style wst/toolbar-title-icon
       :color :white}]]])
 
 (defn toolbar-buttons []
-  [rn/view {:style st/toolbar-buttons-container}
-   [vi/touchable-icon :icons/dots_vertical {:color :white} show-not-implemented!]
-   [vi/touchable-icon :icons/qr {:color :white} show-not-implemented!]])
+  [rn/view {:style wst/toolbar-buttons-container}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/main/views.cljs]: wst says nothing, better to use wallet-styles

File: react-native/src/status_im/react_native/resources.cljs [hunk 12]
```diff
@@ -15,3 +15,6 @@
    :ethlance           (js/require "./resources/images/contacts/ethlance.png")
    :commiteth          (js/require "./resources/images/contacts/commiteth.png")
    :etherplay          (js/require "./resources/images/contacts/etherplay.png")})
+
+(def assets
```
Comment (flexsurfer) [inline react-native/src/status_im/react_native/resources.cljs:19]: imo  if we can't use svg icons we should use native images, because on iphones this image will be looking bad

File: src/status_im/components/list/views.cljs (hunk 8 above)
Comment (jeluard) [reply src/status_im/components/list/views.cljs]: Weird

File: src/status_im/components/toolbar_new/view.cljs (hunk 6 above)
Comment (jeluard) [reply src/status_im/components/toolbar_new/view.cljs:88]: TooManyToolbarsExceptions.

`toolbar2` should disappear soon.

File: src/status_im/ui/screens/discover/views/search_results.cljs (hunk 10 above)
Comment (jeluard) [reply src/status_im/ui/screens/discover/views/search_results.cljs]: ok

File: src/status_im/ui/screens/wallet/main/views.cljs (hunk 11 above)
Comment (jeluard) [reply src/status_im/ui/screens/wallet/main/views.cljs]: ok

File: src/status_im/components/toolbar_new/view.cljs (hunk 6 above)
Comment (oskarth) [reply src/status_im/components/toolbar_new/view.cljs:88]: Can we document this with DEPRECATED and a short comment?

Comment (jeluard) [reply src/status_im/components/toolbar_new/view.cljs:88]: sure

File: react-native/src/status_im/react_native/resources.cljs (hunk 12 above)
Comment (flexsurfer) [reply react-native/src/status_im/react_native/resources.cljs:19]: but maybe it's ok for now

File: src/status_im/ui/screens/wallet/main/views.cljs (hunk 11 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/main/views.cljs]: thanks

**Review by x-goose-x — APPROVED** [review]
